File: model/src/test/java/pl/agh/harmonytools/model/ChordComponentManagerTest.java
package pl.agh.harmonytools.model;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import pl.agh.harmonytools.model.util.ChordComponentManager;

public class ChordComponentManagerTest {

    @Test
    public void instancesCheck() {
        ChordComponent cc1 = ChordComponentManager.chordComponentFromString("5>", false);
        ChordComponent cc2 = ChordComponentManager.chordComponentFromString("5>", false);
        Assertions.assertEquals(cc1, cc2);

        cc1 = ChordComponentManager.chordComponentFromString("5>", true);
        cc2 = ChordComponentManager.chordComponentFromString("5>", false);
        Assertions.assertNotEquals(cc1, cc2);

        cc1 = ChordComponentManager.chordComponentFromString("5>", true);
        cc2 = ChordComponentManager.chordComponentFromString("5>", true);
        Assertions.assertEquals(cc1, cc2);
    }
}

File: model/src/test/scala/pl/agh/harmonytools/model/util/ChordComponentManagerTest.scala
package pl.agh.harmonytools.model.util

import org.scalatest.{FunSuite, Matchers}

class ChordComponentManagerTest extends FunSuite with Matchers {
  test("instances check") {
    val cc1 = ChordComponentManager.chordComponentFromString("5>")
    val cc2 = ChordComponentManager.chordComponentFromString("5>")
    cc1 shouldBe cc2
    val cc3 = ChordComponentManager.chordComponentFromString("5>", isDown = true)
    val cc4 = ChordComponentManager.chordComponentFromString("5>")
    cc3 should not be cc4
    val cc5 = ChordComponentManager.chordComponentFromString("5>", isDown = true)
    val cc6 = ChordComponentManager.chordComponentFromString("5>", isDown = true)
    cc5 shouldBe cc6
  }
}
